layers/+lang/ocaml/funcs.el

```diff
@@ -23,26 +23,30 @@
 
 (defun spacemacs//init-ocaml-opam ()
   (if (executable-find "opam")
-      (let ((share (string-trim-right
-                    (with-output-to-string
-                      (with-current-buffer
-                          standard-output
-                        (process-file
-                         shell-file-name nil '(t nil) nil shell-command-switch
-                         "opam config var share"))))))
-        (cond ((string= "" share)
-               (spacemacs-buffer/warning
-                "\"opam config var share\" output empty string."))
-              ((not (file-directory-p share))
-               (spacemacs-buffer/warning
-                "opam share directory does not exist."))
-              (t (setq opam-share share
-                       opam-load-path (concat share "/emacs/site-lisp"))
-                 (add-to-list 'load-path opam-load-path))))
+      (cl-flet ((share-dir-from-command (command)
```

Comment:
can you combine `cl-flet` and `let*` forms and replace them by `cl-letf*`?
that kill remove one level of nesting.

Reply:
ended up expanding the `let*` to include `share-dir-from-command` as a lambda variable and then called it using `funcall`, which seems clearer than `letf` (or my original solution for that matter). if this is unidiomatic or what-have-you i can redo.